api/v1alpha1/sleepinfo_types.go

```diff
@@ -130,24 +159,49 @@ type SleepInfo struct {
 }
 
 func (s SleepInfo) GetSleepSchedule() (string, error) {
-	return s.getScheduleFromWeekdayAndTime(s.Spec.SleepTime)
+	return s.getScheduleFromWeekdayAndTime(s.Spec.WeekDaySleep, s.Spec.SleepTime)
 }
 
 func (s SleepInfo) GetWakeUpSchedule() (string, error) {
 	if s.Spec.WakeUpTime == "" {
 		return "", nil
 	}
-	return s.getScheduleFromWeekdayAndTime(s.Spec.WakeUpTime)
+	return s.getScheduleFromWeekdayAndTime(s.Spec.WeekDayWakeUp, s.Spec.WakeUpTime)
 }
 
 func (s SleepInfo) GetExcludeRef() []ExcludeRef {
 	return s.Spec.ExcludeRef
 }
 
-func (s SleepInfo) getScheduleFromWeekdayAndTime(hourAndMinute string) (string, error) {
-	weekday := s.Spec.Weekdays
+func (s SleepInfo) GetScheduleException() ([]string, error) {
+	scheduleExceptions := []string{}
+	for _, exception := range s.Spec.ScheduleException {
+		splittedDate := strings.Split(exception.Date, "-")
+		//nolint:gomnd
+		if len(splittedDate) != 2 {
+			return nil, fmt.Errorf("date should be of format MM-DD, actual: '%s'", exception.Date)
+		}
+		splittedTime := strings.Split(exception.SleepAt, ":")
+		//nolint:gomnd
+		if len(splittedTime) != 2 {
+			return nil, fmt.Errorf("time should be of format HH:mm, actual: '%s'", exception.SleepAt)
+		}
+		schedule := fmt.Sprintf("%s %s %s %s *", splittedTime[1], splittedTime[0], splittedDate[0], splittedDate[1])
+		if s.Spec.TimeZone != "" {
+			schedule = fmt.Sprintf("CRON_TZ=%s %s", s.Spec.TimeZone, schedule)
+		}
+
+		scheduleExceptions = append(scheduleExceptions, schedule)
+	}
+	return scheduleExceptions, nil
+}
+
+func (s SleepInfo) getScheduleFromWeekdayAndTime(weekday string, hourAndMinute string) (string, error) {
```

Comment:
We could make this function more generic to be reused in the GetScheduleException one?